tests/autoscaler/autoscaler_test.go

```diff
@@ -0,0 +1,203 @@
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/tests/ctrlclient"
+)
+
+const (
+	helloWorldNamespace      = "default"
+	helloWorldDeploymentName = "helloworld"
+)
+
+func Test_Autoscaler(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	tcCtrlClient, err := ctrlclient.CreateTCCtrlClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating TC k8s client: %v", err)
+	}
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logger.Debugf(ctx, "Creating %s deployment", helloWorldDeploymentName)
+
+	deployment, err := createDeployment(ctx, tcCtrlClient, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Get number of worker nodes.
+	workersCount, err := getWorkersCount(ctx, tcCtrlClient)
+	if err != nil {
+		cleanupAndFatal(ctx, deployment, tcCtrlClient, t, "%v", err)
+	}
+
+	logger.Debugf(ctx, "Found %d worker nodes", workersCount)
+
+	// Scale helloworld deployment to len(workers) + 1 replicas to trigger a scale up.
+	expectedWorkersCount := int32(workersCount + 1)
+	logger.Debugf(ctx, "Scaling deployment %s/%s to %d replicas", helloWorldNamespace, helloWorldDeploymentName, expectedWorkersCount)
+	err = scaleDeployment(ctx, tcCtrlClient, expectedWorkersCount)
+	if err != nil {
+		cleanupAndFatal(ctx, deployment, tcCtrlClient, t, "%v", err)
+	}
+
+	logger.Debugf(ctx, "Waiting for %d worker nodes to exist", expectedWorkersCount)
+
+	// Wait for nodes to increase by one.
+	o := func() error {
+		workersCount, err := getWorkersCount(ctx, tcCtrlClient)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if int32(workersCount) != expectedWorkersCount {
+			return microerror.Maskf(executionFailedError, "Expecting %d workers, %d found", expectedWorkersCount, workersCount)
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(backoff.LongMaxWait, 10*time.Second)
+	n := backoff.NewNotifier(logger, ctx)
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		cleanupAndFatal(ctx, deployment, tcCtrlClient, t, "timeout waiting for cluster to scale up: %v", err)
+	}
+
+	// Scale down deployment, wait for one node to get deleted.
+	expectedWorkersCount = expectedWorkersCount - 1
+	logger.Debugf(ctx, "Scaling deployment %s/%s to %d replicas", helloWorldNamespace, helloWorldDeploymentName, expectedWorkersCount)
+	err = scaleDeployment(ctx, tcCtrlClient, expectedWorkersCount)
+	if err != nil {
+		cleanupAndFatal(ctx, deployment, tcCtrlClient, t, "timeout waiting for cluster to scale down: %v", err)
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		cleanupAndFatal(ctx, deployment, tcCtrlClient, t, "timeout waiting for cluster to scale down: %v", err)
+	}
+}
```

Comment:
Now that there's no `defer`, I think you need to have the `Delete()` for deployment here in the end again :slightly_smiling_face:

Reply:
true, added